packages/terra-paginator/src/Paginator.jsx

```diff
@@ -0,0 +1,229 @@
+      pageSequence: selectedPage && totalCount ? pageSet(selectedPage, calculatePages(totalCount, itemCountPerPage)) : undefined,
+    };
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.hasNavContext = this.hasNavContext.bind(this);
+    this.buildPageButtons = this.buildPageButtons.bind(this);
+    this.reducedPaginator = this.reducedPaginator.bind(this);
+  }
+
+  handlePageChange(index) {
+    return (event) => {
+      event.preventDefault();
+
+      if (isNaN(index)) {
+        this.props.onPageChange(event.target.text.trim().toLowerCase());
+
+        return false;
+      }
+
+      this.props.onPageChange(index);
+      this.setState({
+        selectedPage: index,
+        pageSequence: pageSet(index, calculatePages(this.props.totalCount, this.props.itemCountPerPage)),
+      });
+
+      return false;
+    };
+  }
+
+  handleOnKeyDown(index) {
+    return (event) => {
+      if (event.nativeEvent.keyCode === KEYCODES.ENTER || event.nativeEvent.keyCode === KEYCODES.SPACE) {
+        if (isNaN(index)) {
+          this.props.onPageChange(event.target.text.trim().toLowerCase());
+
+          return false;
+        }
+
+        this.props.onPageChange(index);
+        this.setState({
+          selectedPage: index,
+          pageSequence: pageSet(index, calculatePages(this.props.totalCount, this.props.itemCountPerPage)),
+        });
+      }
+
+      return false;
+    };
+  }
+
+  buildPageButtons(totalPages, onClick) {
+    const { pageSequence, selectedPage } = this.state;
+
+    const pageButtons = [];
+
+    pageSequence.forEach((val) => {
+      const paginationLinkClassNames = cx([
+        'nav-link',
+        val === selectedPage ? 'is-selected' : null,
+      ]);
+
+      if (val > totalPages) {
+        return;
+      }
+      /* eslint-disable comma-dangle */
+      pageButtons.push(
+        <a
+          className={paginationLinkClassNames}
+          tabIndex={val === selectedPage ? null : '0'}
+          key={`pageButton_${val}`}
+          onClick={onClick(val)}
+          onKeyDown={this.handleOnKeyDown(val)}
+        >
+          {val}
+        </a>
+      );
+      /* eslint-enable comma-dangle */
+    });
+
+    return pageButtons;
+  }
+
+  hasNavContext() {
+    return this.props.totalCount && this.props.itemCountPerPage;
+  }
+
+  defaultPaginator() {
+    const totalPages = calculatePages(this.props.totalCount, this.props.itemCountPerPage);
+    const { selectedPage } = this.state;
+    const previousPageIndex = selectedPage === 1 ? 1 : selectedPage - 1;
+    const nextPageIndex = selectedPage === totalPages ? totalPages : selectedPage + 1;
+
+    const fullView = (
+      <div className={cx(['paginator', !this.hasNavContext() && 'pageless'])}>
+        {
+          this.hasNavContext() && <a
+            className={cx(['nav-link', 'left-controls', selectedPage === 1 && 'is-disabled'])}
+            tabIndex="0"
```

Comment:
Discovered links that were disabled were still getting tab-focus even though styles weren't showing it. Need to update: `tabIndex="0"` to 
`tabIndex={selectedPage === 1 ? null : '0'}`

Reply:
https://github.com/cerner/terra-core/pull/1445/commits/37187bdacdbd135aef8fab5bf1ca8b24cbbc2bf9